app/src/main/java/tech/ula/MainActivity.kt

```diff
@@ -170,62 +171,54 @@ class MainActivity : AppCompatActivity(), SessionListFragment.SessionSelection,
         ACRA.init(applicationContext as Application, builder)
     }
 
-    private fun setupReviewRequest() {
-        val minimumTimePassedReached = usageUtility.getIsSufficientTimeElapsedSinceFirstOpen()
-        val minimumTimesOpenedReached = usageUtility.numberOfTimesOpenedIsGreaterThanThreshold()
-        val userAlreadyGaveReviewOrDismissed = usageUtility.getUserGaveFeedback()
+    private fun setupReviewRequestUI() {
+        val requestReviewView = layoutInflater.inflate(R.layout.list_item_review_request, null)
+        val viewHolder = findViewById<ViewGroup>(R.id.request_review_insert_point)
 
-        if (minimumTimePassedReached && minimumTimesOpenedReached && !userAlreadyGaveReviewOrDismissed) {
-            val requestReviewView = layoutInflater.inflate(R.layout.list_item_review_request, null)
-            requestReviewView.visibility = View.VISIBLE
-            val viewHolder = findViewById<ViewGroup>(R.id.request_review_insert_point)
-            viewHolder.addView(requestReviewView, 0)
-            setupReviewRequestUI(viewHolder)
-        }
-    }
+        requestReviewView.visibility = View.VISIBLE
+        viewHolder.addView(requestReviewView, 0)
```

Comment:
I'll preface this by saying that views, view groups, layout inflation etc. is still one of the Android areas I feel weakest on. It looks like this could be changed to potentially be more performant, and avoid the IDE warning that using `null` in the call to `inflate` gives.

If you remove the `android:visibility` attribute from `list_item_review_request`, initialize the `viewHolder` first, you can then use the `viewHolder` as the `root` parameter to the `inflate` call. You'd then be able to remove the following two lines that set visibility and add the view the the `viewHolder`. Then in `handleUserFeedback`, you can call `viewHolder.removeAllViews()` instead of adjusting the visibility. I think this should reduce memory footprint, and if I remember right views are one of the more costly pieces of Android.

Reply:
That's a good idea.  I did notice an improvement in performance when navigating to different fragments when switching to this approach